File: Selenium/Activity17.java
package Selenium;

import java.util.List;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.support.ui.Select;

public class Activity17 {

	public static void main(String[] args) {
		WebDriver driver = new ChromeDriver();
		driver.manage().window().maximize();
		driver.get("https://training-support.net/webelements/selects");
		System.out.println("Page Title is : " + driver.getTitle());
		WebElement dropDown = driver.findElement(By.cssSelector("select.h-80"));
		Select select = new Select(dropDown);
		select.selectByVisibleText("HTML");
		for(int i=3; i<=5;i++) {
			select.selectByIndex(i);
		}
		
		select.selectByValue("nodejs");
		List<WebElement> options = select.getAllSelectedOptions();
		System.out.println("-----------All Selected options are----------------");
		for(WebElement option : options) {
			System.out.println(option.getText());
		}
		select.deselectByIndex(4);
		options = select.getAllSelectedOptions();
		System.out.println("-----------All Selected options are----------------");
		for(WebElement option : options) {
			System.out.println(option.getText());
		}
		driver.quit();
	}

}

File: Selenium/Activity17.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.select import Select

with webdriver.Chrome() as driver:
    driver.get("https://training-support.net/webelements/selects")
    print(driver.title)
    dropdown = driver.find_element(By.CSS_SELECTOR, "select.h-80")
    opt = Select(dropdown)
    opt.select_by_visible_text("HTML")
    print("By viisble text : " , opt.first_selected_option.text)
    for i in range(3,5) :
        opt.select_by_index(i)
    opt.select_by_value("nodejs")
    ddOptions = opt.all_selected_options
    print("-----------All Selected options are----------------")
    for ddopt in ddOptions:
        print(ddopt.text)
    opt.deselect_by_index(4)
    ddOptions = opt.all_selected_options
    print("-----------All Selected options are----------------")
    for ddopt in ddOptions:
        print(ddopt.text)
    driver.quit()